### ISSUE TITLE
Why is there no Item versus ItemT?

### ISSUE THREAD

**Issue by davegi** [louthy/language-ext#350]
I'm trying to use `ItemT.satisfy()` as follows but the result generates an error when I try and use it in a LINQ `from` clause:

`private static LanguageExt.Parsec.Parser<string, string> ColorParser(string color) => LanguageExt.Parsec.ItemT.satisfy<string>(_ => IsColor(color));
`

                    this.Parser =
                    from ...
                    from ...
                    from color in ColorParser(colorString) // CS1943
                    from ...
                    from ...
                    select constructor(color);

> CS1943	An expression of type 'Parser<string, string>' is not allowed in a subsequent from clause in a query expression with source type 'Parser<<anonymous type: string ..., string ...>>'.  Type inference failed in the call to 'SelectMany'.

Obviously I have a type mismatch between `Parser<string>` and `Parser<string, string>` which makes sense since I'm using `ItemT`.

Bottom line:
- Can I convert a `Parser<I, O>` to `Parser<I>`?
- Is my approach correct but I should be using `Item`, which doesn't seem to exist?
- Is there another approach that I'm not aware of (functional and lang-ext are very new to me)?

Thanks,
djg

**Comment by louthy** [comment]
On my phone, so can’t go in depth, but you can convert a Parser<I, O> to a
Parser<A> as long as the I is a char. You can call ToParser() to convert.

**Comment by davegi** [comment]
Thanks Paul. I should have mentioned that I found this method but it doesn't seem to apply in my case since `I` is not a `char`...or am I missing some functional / lang-ext (magical incantation;-) way of gluing these pieces together?

**Comment by louthy** [comment]
@davegi Not really, no.  The `Parser<A>` essentially is a `Parser<char, A>` because it parses a stream of `char`.  So, unless you can map your `I` input stream to `char` then there is no type compatible way of using them together.
